### PR TITLE
cpu/stm32l1: optimized and fixed timer driver

### PR DESCRIPTION
**Pull request by haukepetersen**
saves ~600 byte ROM and looks nicer :-)

also #3525 works nicely with this.

### CODE REVIEW

File: boards/nucleo-l1/include/periph_conf.h [hunk 1]
```diff
@@ -45,37 +47,14 @@ extern "C" {
...
-#define TIMER_1_IRQ_PRIO    1
-#define TIMER_1_TRIG_SEL    TIM_SMCR_TS_1
+static const timer_conf_t timer_config[] = {
+    /* device, RCC bit, IRQ bit */
+    {TIM5, 3, TIM5_IRQn},
+};
+/* interrupt routines */
+#define TIMER_0_ISR         (isr_tim5)
+/* number of defined timers */
+#define TIMER_NUMOF         (sizeof(timer_config) / sizeof(timer_config[0]))
```
Comment (A-Paul) [inline boards/nucleo-l1/include/periph_conf.h]: A division for every time referencing a constant value?

Comment (gebart) [inline boards/nucleo-l1/include/periph_conf.h]: The constant will be computed by the compiler at compile time, there won't
be a division at run time.

Comment (haukepetersen) [inline boards/nucleo-l1/include/periph_conf.h]: exactly :-)

**Comment by haukepetersen** [comment]
adjusted configuration for the `limifrog-l1`. @PeterKietzmann do you have this board for testing this?

**Comment by PeterKietzmann** [comment]
@ReneHerthel took it home. So he has to. I'll also send him a private mail

**Comment by haukepetersen** [comment]
thx

**Comment by ReneHerthel** [comment]
Okey, I cloned this branch and tested it with the xtimer-test applications from RIOT.
`tests/xtimer_msg` -> every second a new output like 'sec=32 min=0 hour=0" ... "sec=33..."
`tests/xtimer_remove` -> "Test successful."
`tests/xtimer_reset` -> "Test completed!"
`tests/xtimer_usleep_until` -> "Test complete."
`tests/xtimer_shift_on_compare` -> I got an error "The selected BOARD=limifrog-v1 is not whitelisted:"
Don't know if the last one is crucial.

**Comment by OlegHahm** [comment]
> tests/xtimer_shift_on_compare -> I got an error "The selected BOARD=limifrog-v1 is not whitelisted:"
Don't know if the last one is crucial.

I think this is okay, because this test is only for platforms with timers smaller than 32 bit.

**Comment by authmillenon** [comment]
Needs rebase

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
@OlegHahm, care to give this a fix test-run?

**Comment by OlegHahm** [comment]
Ah, now I remember. I cannot test, because I don't any output on this board.

**Comment by haukepetersen** [comment]
If I tell you it's working fine here, does that suffice?

**Comment by haukepetersen** [comment]
Or you can come over and I give you a live demo (as live demos always work...) :-)

**Comment by OlegHahm** [comment]
Okay, got it working with the second UART, but `xtimer_set_msg` is failing:
```
2015-09-21 17:21:16,270 - INFO # main(): This is RIOT! (Version: 2014.12-3522-g968b3-tbilisi)
2015-09-21 17:21:16,270 - INFO # This is thread 3
2015-09-21 17:21:16,271 - INFO # sending 1st msg
2015-09-21 17:21:16,271 - INFO # now=0:262946 -> every 2.0s: Hello World
2015-09-21 17:21:16,272 - INFO # sending 2nd msg
2015-09-21 17:21:16,272 - INFO # now=0:419176 -> every 5.0s: This is a Test
2015-09-21 17:21:16,273 - INFO # This is thread 4
2015-09-21 17:21:16,302 - INFO # sec=1 min=0 hour=0
2015-09-21 17:21:16,329 - INFO # now=2:375343 -> every 2.0s: Hello World
2015-09-21 17:21:16,335 - INFO # sec=2 min=0 hour=0
2015-09-21 17:21:16,368 - INFO # sec=3 min=0 hour=0
2015-09-21 17:21:16,395 - INFO # now=4:487744 -> every 2.0s: Hello World
2015-09-21 17:21:16,400 - INFO # sec=4 min=0 hour=0
2015-09-21 17:21:16,428 - INFO # now=5:540153 -> every 5.0s: This is a Test
2015-09-21 17:21:16,433 - INFO # sec=5 min=0 hour=0
2015-09-21 17:21:16,460 - INFO # now=6:600145 -> every 2.0s: Hello World
2015-09-21 17:21:16,466 - INFO # sec=6 min=0 hour=0
2015-09-21 17:21:16,499 - INFO # sec=7 min=0 hour=0
2015-09-21 17:21:16,527 - INFO # now=8:712545 -> every 2.0s: Hello World
2015-09-21 17:21:16,532 - INFO # sec=8 min=0 hour=0
2015-09-21 17:21:16,565 - INFO # sec=10 min=0 hour=0
2015-09-21 17:21:16,588 - INFO # now=10:661060 -> every 5.0s: This is a Test
2015-09-21 17:21:16,593 - INFO # now=10:824948 -> every 2.0s: Hello World
2015-09-21 17:21:16,598 - INFO # sec=11 min=0 hour=0
2015-09-21 17:21:16,631 - INFO # sec=12 min=0 hour=0
2015-09-21 17:21:16,659 - INFO # now=12:940122 -> every 2.0s: Hello World
2015-09-21 17:21:16,663 - INFO # sec=13 min=0 hour=0
2015-09-21 17:21:16,696 - INFO # sec=14 min=0 hour=0
2015-09-21 17:21:16,724 - INFO # now=15:55293 -> every 2.0s: Hello World
2015-09-21 17:21:16,729 - INFO # sec=15 min=0 hour=0
2015-09-21 17:21:16,748 - INFO # now=15:784725 -> every 5.0s: This is a Test
2015-09-21 17:21:16,762 - INFO # sec=16 min=0 hour=0
2015-09-21 17:21:16,790 - INFO # now=17:167701 -> every 2.0s: Hello World
2015-09-21 17:21:16,795 - INFO # sec=17 min=0 hour=0
2015-09-21 17:21:16,828 - INFO # sec=18 min=0 hour=0
2015-09-21 17:21:16,856 - INFO # now=19:282869 -> every 2.0s: Hello World
2015-09-21 17:21:16,861 - INFO # sec=19 min=0 hour=0
2015-09-21 17:21:16,894 - INFO # sec=20 min=0 hour=0
2015-09-21 17:21:16,907 - INFO # now=20:908393 -> every 5.0s: This is a Test
2015-09-21 17:21:16,922 - INFO # now=21:398041 -> every 2.0s: Hello World
2015-09-21 17:21:16,926 - INFO # sec=21 min=0 hour=0
2015-09-21 17:21:16,959 - INFO # sec=22 min=0 hour=0
2015-09-21 17:21:16,988 - INFO # now=23:513210 -> every 2.0s: Hello World
2015-09-21 17:21:16,992 - INFO # sec=23 min=0 hour=0
2015-09-21 17:21:17,025 - INFO # sec=24 min=0 hour=0
2015-09-21 17:21:17,053 - INFO # now=25:628387 -> every 2.0s: Hello World
2015-09-21 17:21:17,058 - INFO # sec=25 min=0 hour=0
2015-09-21 17:21:17,067 - INFO # now=26:32065 -> every 5.0s: This is a Test
2015-09-21 17:21:17,090 - INFO # sec=26 min=0 hour=0
2015-09-21 17:21:17,119 - INFO # now=27:743558 -> every 2.0s: Hello World
2015-09-21 17:21:17,123 - INFO # sec=27 min=0 hour=0
2015-09-21 17:21:17,156 - INFO # sec=29 min=0 hour=0
2015-09-21 17:21:17,185 - INFO # now=29:858729 -> every 2.0s: Hello World
2015-09-21 17:21:17,189 - INFO # sec=30 min=0 hour=0
2015-09-21 17:21:17,226 - INFO # sec=31 min=0 now=31:152986 -> every 5.0s: This is a Test
2015-09-21 17:21:17,251 - INFO # h-> evenow=31:973904 -> every 2.0s: Hello World
2015-09-21 17:21:17,259 - INFO # sec=32 min=0 hour=0
2015-09-21 17:21:17,292 - INFO # sec=33 min=0 hour=0
2015-09-21 17:21:17,317 - INFO # now=34:89077 -> every 2.0s: Hello World
2015-09-21 17:21:17,325 - INFO # sec=34 min=0 hour=0
2015-09-21 17:21:17,358 - INFO # sec=35 min=0 hour=0
2015-09-21 17:21:17,387 - INFO # now=36:201477 -> every 2.0s: Hello World
2015-09-21 17:21:17,389 - INFO # now=36:316168 -> every 5.0s: This is a Test
2015-09-21 17:21:17,390 - INFO # sec=36 min=0 hour=0
2015-09-21 17:21:17,423 - INFO # sec=37 min=0 hour=0
2015-09-21 17:21:17,449 - INFO # now=38:316646 -> every 2.0s: Hello World
2015-09-21 17:21:17,456 - INFO # sec=38 min=0 hour=0
2015-09-21 17:21:17,489 - INFO # sec=39 min=0 hour=0
2015-09-21 17:21:17,515 - INFO # now=40:431823 -> every 2.0s: Hello World
2015-09-21 17:21:17,522 - INFO # sec=40 min=0 hour=0
2015-09-21 17:21:17,547 - INFO # now=41:439912 -> every 5.0s: This is a Test
2015-09-21 17:21:17,555 - INFO # sec=41 min=0 hour=0
2015-09-21 17:21:17,581 - INFO # now=42:546992 -> every 2.0s: Hello World
2015-09-21 17:21:17,588 - INFO # sec=42 min=0 hour=0
2015-09-21 17:21:17,621 - INFO # sec=43 min=0 hour=0
2015-09-21 17:21:17,647 - INFO # now=44:662162 -> every 2.0s: Hello World
2015-09-21 17:21:17,654 - INFO # sec=44 min=0 hour=0
2015-09-21 17:21:17,686 - INFO # sec=46 min=0 hour=0
2015-09-21 17:21:17,706 - INFO # now=46:563578 -> every 5.0s: This is a Test
2015-09-21 17:21:17,713 - INFO # now=46:777330 -> every 2.0s: Hello World
2015-09-21 17:21:17,719 - INFO # sec=47 min=0 hour=0
2015-09-21 17:21:17,752 - INFO # sec=48 min=0 hour=0
2015-09-21 17:21:17,778 - INFO # now=48:892509 -> every 2.0s: Hello World
2015-09-21 17:21:17,785 - INFO # sec=49 min=0 hour=0
2015-09-21 17:21:17,818 - INFO # sec=50 min=0 hour=0
2015-09-21 17:21:17,844 - INFO # now=51:7682 -> every 2.0s: Hello World
2015-09-21 17:21:17,851 - INFO # sec=51 min=0 hour=0
2015-09-21 17:21:17,866 - INFO # now=51:687254 -> every 5.0s: This is a Test
2015-09-21 17:21:17,884 - INFO # sec=52 min=0 hour=0
2015-09-21 17:21:17,910 - INFO # now=53:117310 -> every 2.0s: Hello World
2015-09-21 17:21:17,917 - INFO # sec=53 min=0 hour=0
2015-09-21 17:21:17,949 - INFO # sec=54 min=0 hour=0
2015-09-21 17:21:17,976 - INFO # now=55:232486 -> every 2.0s: Hello World
2015-09-21 17:21:17,982 - INFO # sec=55 min=0 hour=0
2015-09-21 17:21:18,015 - INFO # sec=56 min=0 hour=0
2015-09-21 17:21:18,025 - INFO # now=56:810921 -> every 5.0s: This is a Test
2015-09-21 17:21:18,042 - INFO # now=57:347654 -> every 2.0s: Hello World
2015-09-21 17:21:18,048 - INFO # sec=57 min=0 hour=0
2015-09-21 17:21:18,081 - INFO # sec=58 min=0 hour=0
2015-09-21 17:21:18,108 - INFO # now=59:462824 -> every 2.0s: Hello World
2015-09-21 17:21:18,114 - INFO # sec=59 min=0 hour=0
2015-09-21 17:21:18,147 - INFO # sec=60 min=1 hour=0
2015-09-21 17:21:18,174 - INFO # now=61:577994 -> every 2.0s: Hello World
2015-09-21 17:21:18,179 - INFO # sec=61 min=1 hour=0
2015-09-21 17:21:18,185 - INFO # now=61:934591 -> every 5.0s: This is a Test
2015-09-21 17:21:18,212 - INFO # sec=62 min=1 hour=0
2015-09-21 17:21:18,239 - INFO # now=63:693174 -> every 2.0s: Hello World
2015-09-21 17:21:18,245 - INFO # sec=63 min=1 hour=0
2015-09-21 17:21:18,278 - INFO # sec=65 min=1 hour=0
2015-09-21 17:21:18,305 - INFO # now=65:808346 -> every 2.0s: Hello World
2015-09-21 17:21:18,311 - INFO # sec=66 min=1 hour=0
2015-09-21 17:21:18,346 - INFO # now=67:58261 -> every 5.0s: This is a Test
2015-09-21 17:21:18,347 - INFO # sec=67 min=1 hour=0
2015-09-21 17:21:18,371 - INFO # now=67:923519 -> every 2.0s: Hello World
2015-09-21 17:21:18,378 - INFO # sec=68 min=1 hour=0
2015-09-21 17:21:18,411 - INFO # sec=69 min=1 hour=0
2015-09-21 17:21:18,437 - INFO # now=70:38682 -> every 2.0s: Hello World
2015-09-21 17:21:18,444 - INFO # sec=70 min=1 hour=0
2015-09-21 17:21:18,477 - INFO # sec=71 min=1 hour=0
2015-09-21 17:21:18,507 - INFO # now=72:151093 -> every 2.0s: Hello World
2015-09-21 17:21:18,508 - INFO # now=72:265786 -> every 5.0s: This is a Test
2015-09-21 17:21:18,510 - INFO # sec=72 min=1 hour=0
2015-09-21 17:21:18,543 - INFO # sec=73 min=1 hour=0
2015-09-21 17:21:18,569 - INFO # now=74:266264 -> every 2.0s: Hello World
2015-09-21 17:21:18,576 - INFO # sec=74 min=1 hour=0
2015-09-21 17:21:18,609 - INFO # sec=75 min=1 hour=0
2015-09-21 17:21:18,635 - INFO # now=76:381431 -> every 2.0s: Hello World
2015-09-21 17:21:18,641 - INFO # sec=76 min=1 hour=0
2015-09-21 17:21:18,666 - INFO # now=77:389522 -> every 5.0s: This is a Test
2015-09-21 17:21:18,674 - INFO # sec=77 min=1 hour=0
2015-09-21 17:21:18,700 - INFO # now=78:496601 -> every 2.0s: Hello World
2015-09-21 17:21:18,707 - INFO # sec=78 min=1 hour=0
2015-09-21 17:21:18,740 - INFO # sec=79 min=1 hour=0
2015-09-21 17:21:18,766 - INFO # now=80:611780 -> every 2.0s: Hello World
2015-09-21 17:21:18,773 - INFO # sec=80 min=1 hour=0
2015-09-21 17:21:18,806 - INFO # sec=81 min=1 hour=0
2015-09-21 17:21:18,826 - INFO # now=82:513198 -> every 5.0s: This is a Test
2015-09-21 17:21:18,832 - INFO # now=82:726953 -> every 2.0s: Hello World
2015-09-21 17:21:18,838 - INFO # sec=83 min=1 hour=0
2015-09-21 17:21:18,871 - INFO # sec=84 min=1 hour=0
2015-09-21 17:21:18,898 - INFO # now=84:842117 -> every 2.0s: Hello World
2015-09-21 17:21:18,904 - INFO # sec=85 min=1 hour=0
2015-09-21 17:21:18,937 - INFO # sec=86 min=1 hour=0
2015-09-21 17:21:18,964 - INFO # now=86:957292 -> every 2.0s: Hello World
2015-09-21 17:21:18,970 - INFO # sec=87 min=1 hour=0
2015-09-21 17:21:18,986 - INFO # now=87:636866 -> every 5.0s: This is a Test
2015-09-21 17:21:19,003 - INFO # sec=88 min=1 hour=0
2015-09-21 17:21:19,029 - INFO # now=89:72460 -> every 2.0s: Hello World
2015-09-21 17:21:19,035 - INFO # sec=89 min=1 hour=0
2015-09-21 17:21:19,069 - INFO # sec=90 min=1 hour=0
2015-09-21 17:21:19,095 - INFO # now=91:184869 -> every 2.0s: Hello World
2015-09-21 17:21:19,101 - INFO # sec=91 min=1 hour=0
2015-09-21 17:21:19,134 - INFO # sec=92 min=1 hour=0
2015-09-21 17:21:19,144 - INFO # now=92:760535 -> every 5.0s: This is a Test
2015-09-21 17:21:19,161 - INFO # now=93:300042 -> every 2.0s: Hello World
2015-09-21 17:21:19,167 - INFO # sec=93 min=1 hour=0
2015-09-21 17:21:19,200 - INFO # sec=94 min=1 hour=0
2015-09-21 17:21:19,227 - INFO # now=95:415211 -> every 2.0s: Hello World
2015-09-21 17:21:19,233 - INFO # sec=95 min=1 hour=0
2015-09-21 17:21:19,266 - INFO # sec=96 min=1 hour=0
2015-09-21 17:21:19,293 - INFO # now=97:530381 -> every 2.0s: Hello World
2015-09-21 17:21:19,298 - INFO # sec=97 min=1 hour=0
2015-09-21 17:21:19,304 - INFO # now=97:884204 -> every 5.0s: This is a Test
2015-09-21 17:21:19,331 - INFO # sec=98 min=1 hour=0
2015-09-21 17:21:19,359 - INFO # now=99:645553 -> every 2.0s: Hello World
2015-09-21 17:21:19,364 - INFO # sec=99 min=1 hour=0
2015-09-21 17:21:19,397 - INFO # sec=100 min=1 hour=0
2015-09-21 17:21:19,425 - INFO # now=101:760723 -> every 2.0s: Hello World
2015-09-21 17:21:19,430 - INFO # sec=102 min=1 hour=0
2015-09-21 17:21:19,466 - INFO # now=103:7874 -> every 5.0s: This is a Test
2015-09-21 17:21:19,467 - INFO # sec=103 min=1 hour=0
2015-09-21 17:21:19,491 - INFO # now=103:878672 -> every 2.0s: Hello World
2015-09-21 17:21:19,498 - INFO # sec=104 min=1 hour=0
2015-09-21 17:21:19,531 - INFO # sec=105 min=1 hour=0
2015-09-21 17:21:19,557 - INFO # now=105:996613 -> every 2.0s: Hello World
2015-09-21 17:21:19,564 - INFO # sec=106 min=1 hour=0
2015-09-21 17:21:19,597 - INFO # sec=107 min=1 hour=0
2015-09-21 17:21:19,627 - INFO # now=108:114552 -> every 2.0s: Hello World
2015-09-21 17:21:19,628 - INFO # now=108:232021 -> every 5.0s: This is a Test
2015-09-21 17:21:19,630 - INFO # sec=108 min=1 hour=0
2015-09-21 17:21:19,663 - INFO # sec=109 min=1 hour=0
2015-09-21 17:21:19,689 - INFO # now=110:232499 -> every 2.0s: Hello World
2015-09-21 17:21:19,696 - INFO # sec=110 min=1 hour=0
2015-09-21 17:21:19,729 - INFO # sec=111 min=1 hour=0
2015-09-21 17:21:19,754 - INFO # now=112:350434 -> every 2.0s: Hello World
2015-09-21 17:21:19,761 - INFO # sec=112 min=1 hour=0
2015-09-21 17:21:19,786 - INFO # now=113:358533 -> every 5.0s: This is a Test
2015-09-21 17:21:19,794 - INFO # sec=113 min=1 hour=0
2015-09-21 17:21:19,820 - INFO # now=114:468379 -> every 2.0s: Hello World
2015-09-21 17:21:19,827 - INFO # sec=114 min=1 hour=0
2015-09-21 17:21:19,860 - INFO # sec=115 min=1 hour=0
2015-09-21 17:21:19,886 - INFO # now=116:586317 -> every 2.0s: Hello World
2015-09-21 17:21:19,893 - INFO # sec=116 min=1 hour=0
2015-09-21 17:21:19,926 - INFO # sec=117 min=1 hour=0
2015-09-21 17:21:19,946 - INFO # now=118:484967 -> every 5.0s: This is a Test
2015-09-21 17:21:19,952 - INFO # now=118:704266 -> every 2.0s: Hello World
2015-09-21 17:21:19,959 - INFO # sec=119 min=1 hour=0
2015-09-21 17:21:19,992 - INFO # sec=120 min=2 hour=0
2015-09-21 17:21:20,018 - INFO # now=120:822204 -> every 2.0s: Hello World
2015-09-21 17:21:20,025 - INFO # sec=121 min=2 hour=0
2015-09-21 17:21:20,058 - INFO # sec=122 min=2 hour=0
2015-09-21 17:21:20,084 - INFO # now=122:940143 -> every 2.0s: Hello World
2015-09-21 17:21:20,091 - INFO # sec=123 min=2 hour=0
2015-09-21 17:21:20,105 - INFO # now=123:611405 -> every 5.0s: This is a Test
2015-09-21 17:21:20,124 - INFO # sec=124 min=2 hour=0
2015-09-21 17:21:20,150 - INFO # now=125:58092 -> every 2.0s: Hello World
2015-09-21 17:21:20,157 - INFO # sec=125 min=2 hour=0
2015-09-21 17:21:20,190 - INFO # sec=126 min=2 hour=0
2015-09-21 17:21:20,216 - INFO # now=127:173260 -> every 2.0s: Hello World
2015-09-21 17:21:20,223 - INFO # sec=127 min=2 hour=0
2015-09-21 17:21:20,256 - INFO # sec=128 min=2 hour=0
2015-09-21 17:21:20,265 - INFO # now=128:737848 -> every 5.0s: This is a Test
2015-09-21 17:21:20,282 - INFO # now=129:291206 -> every 2.0s: Hello World
2015-09-21 17:21:20,288 - INFO # sec=129 min=2 hour=0
2015-09-21 17:21:20,321 - INFO # sec=130 min=2 hour=0
2015-09-21 17:21:20,348 - INFO # now=131:409144 -> every 2.0s: Hello World
2015-09-21 17:21:20,354 - INFO # sec=131 min=2 hour=0
2015-09-21 17:21:20,387 - INFO # sec=132 min=2 hour=0
2015-09-21 17:21:20,414 - INFO # now=133:527082 -> every 2.0s: Hello World
2015-09-21 17:21:20,425 - INFO # sec=133 min=2 hounow=133:864306 -> every 5.0s: This is a Test
2015-09-21 17:21:20,457 - INFO # revesec=135 min=2 hour=0
2015-09-21 17:21:20,479 - INFO # now=135:645033 -> every 2.0s: Hello World
2015-09-21 17:21:20,490 - INFO # sec=136 min=2 hour=0
2015-09-21 17:21:20,523 - INFO # sec=137 min=2 hour=0
2015-09-21 17:21:20,546 - INFO # now=137:762970 -> every 2.0s: Hello World
2015-09-21 17:21:20,556 - INFO # sec=138 min=2 hour=0
2015-09-21 17:21:20,584 - INFO # now=138:990723 -> every 5.0s: This is a Test
2015-09-21 17:21:20,589 - INFO # sec=139 min=2 hour=0
2015-09-21 17:21:20,612 - INFO # now=139:880911 -> every 2.0s: Hello World
2015-09-21 17:21:20,622 - INFO # sec=140 min=2 hour=0
2015-09-21 17:21:20,655 - INFO # sec=141 min=2 hour=0
2015-09-21 17:21:20,678 - INFO # now=141:998854 -> every 2.0s: Hello World
2015-09-21 17:21:20,688 - INFO # sec=142 min=2 hour=0
2015-09-21 17:21:20,721 - INFO # sec=143 min=2 hour=0
2015-09-21 17:21:20,743 - INFO # now=144:116794 -> every 2.0s: Hello World
```
No further output.

**Comment by OlegHahm** [comment]
Wait, it's weird. Somehow flashing doesn't seem to work.

**Comment by OlegHahm** [comment]
Ok, fixed that, but got the same result for xtimer_set_msg as above.

**Comment by haukepetersen** [comment]
hmm, here it does not stop, though I get the same faulty output...

**Comment by haukepetersen** [comment]
should be fixed now.

**Comment by haukepetersen** [comment]
@OlegHahm: now `xtimer_msg` should also work - the problem was, that the xtimers default configuration did not work...

**Comment by OlegHahm** [comment]
ACK

**Comment by haukepetersen** [comment]
Travis is happy -> and go.

File: cpu/stm32l1/periph/timer.c [hunk 2]
```diff
@@ -119,291 +81,87 @@ int timer_set(tim_t dev, int channel, unsigned int timeout)
...
-        }
-        DEBUG("channel 4 CCR: %08x\n", ((dev1->CCR4<<16) | (0xffff & dev0->CCR4)));
-    }
-    else {
-        dev0->SR = 0;
-    }
-    if (sched_context_switch_request) {
-        thread_yield();
-    }
+    irq_handler(0, timer_config[0].dev);
```
Comment (IldarValiev) [inline cpu/stm32l1/periph/timer.c]: First argument should be "1" here, shouldn't it?

**Pull request merged**